app/src/test/java/org/mozilla/fenix/exceptions/trackingprotection/TrackingProtectionExceptionsInteractorTest.kt

```diff
@@ -4,52 +4,90 @@
+import org.junit.Assert.assertTrue
 import org.junit.Before
 import org.junit.Test
+import org.junit.runner.RunWith
 import org.mozilla.fenix.BrowserDirection
 import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.helpers.FenixRobolectricTestRunner
 import org.mozilla.fenix.settings.SupportUtils
 
+@RunWith(FenixRobolectricTestRunner::class)
 class TrackingProtectionExceptionsInteractorTest {
 
-    @MockK(relaxed = true) private lateinit var activity: HomeActivity
     private lateinit var interactor: TrackingProtectionExceptionsInteractor
 
-    private val results: List<TrackingProtectionException> = emptyList()
+    private val activity: HomeActivity = mockk(relaxed = true)
     private val engine: Engine = mockk(relaxed = true)
-    private val store = BrowserStore()
-    private val capture = CaptureActionsMiddleware<ExceptionsFragmentState, ExceptionsFragmentAction>()
+
+    private val results: List<TrackingProtectionException> = emptyList()
+    private val store = BrowserStore(
+        BrowserState(
+            tabs = listOf(
+                // Using copy to add TP state because `createTab` doesn't support it right now.
+                createTab("https://mozilla.org", false, "tab1")
+                    .copy(trackingProtection = TrackingProtectionState(ignoredOnTrackingProtection = true)),
+                createTab("https://firefox.com", false, "tab2")
+                    .copy(trackingProtection = TrackingProtectionState(ignoredOnTrackingProtection = true))
+            )
+        )
+    )
+    private val capture =
+        CaptureActionsMiddleware<ExceptionsFragmentState, ExceptionsFragmentAction>()
     private val exceptionsStore = ExceptionsFragmentStore(middlewares = listOf(capture))
     private val trackingProtectionUseCases = TrackingProtectionUseCases(store, engine)
-    private val trackingStorage: TrackingProtectionExceptionStorage = mockk(relaxed = true)
+    private val trackingStorage: TrackingProtectionExceptionStorage =
+        object : TrackingProtectionExceptionStorage {
+            override fun fetchAll(onResult: (List<TrackingProtectionException>) -> Unit) {
+                fetchedAll = true
+                onResult(results)
+            }
+
+            override fun removeAll(activeSessions: List<EngineSession>?) {
+                removedAll = true
+            }
+
+            override fun add(session: EngineSession) = Unit
+            override fun contains(session: EngineSession, onResult: (Boolean) -> Unit) = Unit
+            override fun remove(session: EngineSession) = Unit
+            override fun remove(exception: TrackingProtectionException) = Unit
+            override fun restore() = Unit
+        }
+    private val exceptionsItem: (String) -> TrackingProtectionException = {
+        object : TrackingProtectionException {
+            override val url = it
+        }
+    }
+    private var fetchedAll = false
+    private var removedAll = false
```

Comment:
I think we need reset these on `setup()`, as these values will be shared between tests causing flakiness.

Reply:
Yeah, I can add that. Each test is run on a new instance of the test class, so it doesn't affect us today (which is why the tests are not failing), but maybe that will change tomorrow.

Reply:
Heh, you can't use `lateinit` on a primitive:

```diff
diff --git a/app/src/test/java/org/mozilla/fenix/exceptions/trackingprotection/TrackingProtectionExceptionsInteractorTest.kt b/app/src/test/java/org/mozilla/fenix/exceptions/trackingprotection/TrackingProtectionExceptionsInteractorTest.kt
index 8323ef4bd..bf691e271 100644
--- a/app/src/test/java/org/mozilla/fenix/exceptions/trackingprotection/TrackingProtectionExceptionsInteractorTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/exceptions/trackingprotection/TrackingProtectionExceptionsInteractorTest.kt
@@ -76,8 +76,8 @@ class TrackingProtectionExceptionsInteractorTest {
             override val url = it
         }
     }
-    private var fetchedAll = false
-    private var removedAll = false
+    private var fetchedAll: Boolean = false
+    private var removedAll: Boolean = false

     @Before
     fun setup() {
@@ -88,6 +88,9 @@ class TrackingProtectionExceptionsInteractorTest {
         )

         every { engine.trackingProtectionExceptionStore } returns trackingStorage
+
+        fetchedAll = false
+        removedAll = false
     }

     @Test

```